tests/ballerina-integration-test/src/test/java/org/ballerinalang/test/auth/AuthnConfigInheritanceTest.java

```diff
@@ -27,30 +27,122 @@
+        headersMap.put("Authorization", "Basic aXNoYXJhOmFiYw==");
+        HttpResponse response = HttpsClientRequest.doGet(serverInstance.getServiceURLHttps(servicePort, "echo2/test2"),
+                headersMap, serverInstance.getServerHome());
+        Assert.assertNotNull(response);
+        Assert.assertEquals(response.getResponseCode(), 200, "Response code mismatched");
+    }
+
+    @Test(description = "Secured resource, secured service test case with invalid auth headers")
+    public void testInvalidAuthHeaders1() throws Exception {
         Map<String, String> headersMap = new HashMap<>();
         headersMap.put("Authorization", "Basic dGVzdDp0ZXN0MTIz");
-        HttpResponse response = HttpsClientRequest.doGet(serverInstance.getServiceURLHttps(servicePort, "echo/test"),
+        HttpResponse response = HttpsClientRequest.doGet(serverInstance.getServiceURLHttps(servicePort, "echo1/test1"),
                 headersMap, serverInstance.getServerHome());
         Assert.assertNotNull(response);
         Assert.assertEquals(response.getResponseCode(), 401, "Response code mismatched");
     }
+
+    @Test(description = "Non secured resource, secured service test case with invalid auth headers")
+    public void testInvalidAuthHeaders2() throws Exception {
+        Map<String, String> headersMap = new HashMap<>();
+        headersMap.put("Authorization", "Basic dGVzdDp0ZXN0MTIz");
+        HttpResponse response = HttpsClientRequest.doGet(serverInstance.getServiceURLHttps(servicePort, "echo1/test2"),
+                headersMap, serverInstance.getServerHome());
+        Assert.assertNotNull(response);
+        Assert.assertEquals(response.getResponseCode(), 200, "Response code mismatched");
+    }
+
+    @Test(description = "Secured resource, non secured service test case with invalid auth headers")
+    public void testInvalidAuthHeaders3() throws Exception {
+        Map<String, String> headersMap = new HashMap<>();
+        headersMap.put("Authorization", "Basic dGVzdDp0ZXN0MTIz");
+        HttpResponse response = HttpsClientRequest.doGet(serverInstance.getServiceURLHttps(servicePort, "echo2/test1"),
+                headersMap, serverInstance.getServerHome());
+        Assert.assertNotNull(response);
+        Assert.assertEquals(response.getResponseCode(), 401, "Response code mismatched");
+    }
+
+    @Test(description = "Non secured resource, non secured service test case with invalid auth headers")
+    public void testInvalidAuthHeaders4() throws Exception {
+        Map<String, String> headersMap = new HashMap<>();
+        headersMap.put("Authorization", "Basic dGVzdDp0ZXN0MTIz");
+        HttpResponse response = HttpsClientRequest.doGet(serverInstance.getServiceURLHttps(servicePort, "echo2/test2"),
+                headersMap, serverInstance.getServerHome());
+        Assert.assertNotNull(response);
+        Assert.assertEquals(response.getResponseCode(), 200, "Response code mismatched");
```

Comment:
I think we can move the assert statements to a separate function and pass the response and status code for asserting to avoid code duplication. Shall we check in other places as well?